webapp/core/ExportProject.js

```diff
@@ -33,7 +32,7 @@ var ExportProject = function(json, callback){
   };
 
   var _emitError = function(err) {
-    callback({err: err.toString(), status: 400});
+    return Promise.resolve({err: err.toString(), status: 400});
```

Comment:
Almost there. You wrapped as promise-style, but the behavior strategy is the same. Use `resolve` **only** for expected behavior and `reject` for unsatisfied expression or any exception occurred.

In these changes, you are resolving an exception and applying comparison on *successfully* handler [here](https://github.com/franciscovn/terrama2/blob/78a2aa95c344b1ed2fa979b31f76d58520389399/webapp/scripts/ImportProject.js#L29). Ok, the promiser module let you to do it in this way, but it is not best approach for exposing library functionality.

```js
ImportProject(projectCredentials)
  .then((result) => {
    // Success state
  })

  .catch((err) => {
    // Handle Error
  });
```